src/plugins/intel_cpu/src/nodes/mha.cpp

```diff
@@ -0,0 +1,1086 @@
+        jcp.src_prc = dataPrecision == Precision::I8 ? Precision::I32 : Precision::FP32;
+        jcp.dst_prc = dataPrecision;
+        jcp.inner_work_amount = N1;
+        jcp.with_scales = !fqScales2.empty();
+        jcp.broadcast_scales = fqScales2.size() == 1;
+        jcp.src_stride = N1;
+        jcp.dst_stride = batch1 * N1;
+
+        if (mayiuse(cpu_isa_t::avx512_core)) {
+            convertReorderKernel.reset(new jit_convert_reorder_kernel<cpu_isa_t::avx512_core>(jcp));
+        } else if (mayiuse(cpu_isa_t::avx2)) {
+            convertReorderKernel.reset(new jit_convert_reorder_kernel<cpu_isa_t::avx2>(jcp));
+        } else if (mayiuse(cpu_isa_t::sse41)) {
+            convertReorderKernel.reset(new jit_convert_reorder_kernel<cpu_isa_t::sse41>(jcp));
+        } else {
+            THROW_ERROR << "cannot create jit eltwise kernel";
+        }
+    }
+
+    if (mulAddSoftmaxKernel)
+        mulAddSoftmaxKernel->create_ker();
+
+    if (convertReorderKernel)
+        convertReorderKernel->create_ker();
+}
+
+template<typename T>
+static void reorder2D(const T* pin, T* pout, const std::vector<size_t>& dimsOut,
+               const std::vector<size_t>& stridesOut, const std::vector<size_t>& stridesIn) {
+    for (int i0 = 0; i0 < dimsOut[0]; i0++) {
+        for (int i1 = 0; i1 < dimsOut[1]; i1++) {
+            pout[i0 * stridesOut[0] + i1 * stridesOut[1]] = pin[i0 * stridesIn[0] + i1 * stridesIn[1]];
+        }
+    }
+}
+
+void MHA::callBrgemm(brgemmCtx& ctx, std::unique_ptr<brgemm_kernel_t>& brgKernel, const void* pin0, const void* pin1, void* pout, void* wsp) {
+    if (ctx.is_with_amx)
+        amx_tile_configure(ctx.palette);
+    if (ctx.is_with_comp) {
+        brgemm_post_ops_data_t post_ops_data;
+        brgemm_kernel_execute_postops(brgKernel.get(), 1, pin0, pin1, nullptr, pout, pout, post_ops_data, wsp);
+    } else {
+        brgemm_kernel_execute(brgKernel.get(), 1, pin0, pin1, nullptr, pout, wsp);
+    }
+}
+
+template<typename data_type, typename acc_type>
+void MHA::mhaImpl(dnnl::stream strm) {
+    const data_type* pTranspose0In0 = reinterpret_cast<const data_type*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr());
+    const data_type* pTranspose1In0 = reinterpret_cast<const data_type*>(getParentEdgeAt(1)->getMemoryPtr()->GetPtr());
+    const float* pMulIn1 = reinterpret_cast<const float*>(getParentEdgeAt(2)->getMemoryPtr()->GetPtr());
+    const float* pAddIn1 = reinterpret_cast<const float*>(getParentEdgeAt(3)->getMemoryPtr()->GetPtr());
+    const data_type* pTranspose2In0 = reinterpret_cast<const data_type*>(getParentEdgeAt(4)->getMemoryPtr()->GetPtr());
+    data_type* pout = reinterpret_cast<data_type*>(getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+
+    parallel_for2d(dimsMatMul0Out[0], dimsMatMul0Out[1], [&](size_t i0, size_t i1) {
+        size_t threadNum = parallel_get_thread_num();
+
+        auto pTranspose0In0_aux = pTranspose0In0 + i0 * strTranspose0In0[0] + i1 * strTranspose0In0[2]; // order 0213
```

Comment:
The body of parallel section is large. Could we separate it into several inline methods by each step of pipeline if possible? I believe It'd be more readable

Reply:
I agree. Will refactor in separate PR.